internal/project/core_test.go

```diff
@@ -53,3 +58,177 @@ func TestCore_ExistsWithID(t *testing.T) {
 	assert.True(t, ok)
 	assert.NoError(t, err)
 }
+
+func TestCore_Get(t *testing.T) {
+	type fields struct {
+		repo IRepo
+	}
+	type args struct {
+		ctx       context.Context
+		projectID string
+	}
+	ctrl := gomock.NewController(t)
+	mockRepo := mocks.NewMockIRepo(ctrl)
+	ctx := context.Background()
+
+	project1 := &Model{
+		Name:      "test-1",
+		ProjectID: "testID-1",
+		Labels:    map[string]string{"label": "value"},
+	}
+	project2 := &Model{
+		Name:      "test-2",
+		ProjectID: "testID-2",
+		Labels:    map[string]string{"label": "value"},
+	}
+	project3 := &Model{
+		Name:      "test-3",
+		ProjectID: "testID-3",
+		Labels:    map[string]string{"label": "value"},
+	}
+	project1.SetVersion("1")
+	project2.SetVersion("2")
+	project3.SetVersion("3")
+
+	num := rand.Intn(3)
+
+	testProject := &Model{}
+	switch num {
+	case 1:
+		testProject = project1
+	case 2:
+		testProject = project2
+	case 3:
+		testProject = project3
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *Model
+		wantErr bool
+	}{
+		{
+			name: "Get Project Successfully.",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx:       ctx,
+				projectID: testProject.ProjectID,
+			},
+			want:    testProject,
+			wantErr: false,
+		},
+		{
+			name: "Throw error because of Invalid Project ID.",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx:       ctx,
+				projectID: "",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
```

Comment:
this initialisation is same for all tests.. why are we initialising it again..?

Reply:
These are auto-generated from the command I use for generating tests for individual functions. I am only constructing the test cases and the mocks.